homeassistant/components/media_player/sonos.py

```diff
@@ -595,6 +596,8 @@ def update(self):
 
                 if self._queue is None and self.entity_id is not None:
                     self._subscribe_to_player_events()
+            favs = self._player.get_sonos_favorites().get('favorites', [])
```

Comment:
@lwis the method in SoCo actually will always return a result without throwing an exception. In case there are no favorites the array will be in the result but it will be empty (i.e. my check for 'favorites' being present is also not needed).